source/agora/node/Ledger.d

```diff
@@ -59,16 +71,17 @@ public class Ledger
 
     ***************************************************************************/
 
-    public void acceptBlock (const ref Block block) nothrow @safe
+    public void acceptBlock (const ref Block block) @safe
     {
         if (!this.isValidBlock(block))
         {
             logDebug("Rejected block. %s", block);
             return;
         }
 
-        this.ledger ~= block;
-        this.last_block = &this.ledger[$ - 1];
+        this.ledger.saveBlock(block);
+        () @trusted { this.last_block = cast(Block) block; }();
```

Comment:
Why is this cast necessary?

Reply:
Can you suggest a good way?

Reply:
yes, don't cast

Reply:
@Geod24 The issue here was that `this.last_block` was changed from:

```D
const(Block)* last_block;
```

to:

```D
Block last_block;
```

So you can't assign const(Block) to Block.

Reply:
delete last_block then Point to Last Block at block height.